### PR TITLE
add auth scheme integration tests

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: tests/IntegrationTests/IntegrationTestService/Controllers/WeatherForecastController.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@
 namespace IntegrationTestService.Controllers
 {
     [ApiController]
-    [Authorize]
+    [Authorize(AuthenticationSchemes = TestConstants.CustomJwtScheme)]
```
Comment (jmprieur) [inline tests/IntegrationTests/IntegrationTestService/Controllers/WeatherForecastController.cs:17]: Didn't we want to keep the default scheme? given the other controller has another scheme?

File: tests/IntegrationTests/IntegrationTestService/Startup.cs [hunk 2]
```diff
@@ -28,23 +28,32 @@ public void ConfigureServices(IServiceCollection services)
             KeyVaultSecretsProvider _keyVault = new KeyVaultSecretsProvider();
             string ccaSecret = _keyVault.GetSecret(TestConstants.OBOClientKeyVaultUri).Value;
 
-            var builder = services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
-                                  .AddMicrosoftIdentityWebApi(Configuration, subscribeToJwtBearerMiddlewareDiagnosticsEvents: true)
+            services.AddAuthentication()
```
Comment (jmprieur) [inline tests/IntegrationTests/IntegrationTestService/Startup.cs:31]: To have the default scheme, we could call `AddAuthentication(jwtBearerScheme: TestConstants.CustomJwtScheme)` just here too, and then leave the `WeatherForecastController ` controller as it was

File: tests/MultipleAuthSchemes/Controllers/HomeB2CController.cs [hunk 3]
```diff
@@ -24,7 +24,7 @@ public HomeB2CController(ILogger<HomeController> logger, IDownstreamWebApi downs
             ScopeKeySection = "DownstreamB2CApi:Scopes", UserFlow = "b2c_1_susi", AuthenticationScheme = "B2C")]
         public async Task<IActionResult> Index()
         {
-            var value = await _downstreamWebApi.GetForUserAsync<Task>("DownstreamB2CApi", "", null, null, "B2C");
+            var value = await _downstreamWebApi.GetForUserAsync<Task>("DownstreamB2CApi", string.Empty, authenticationScheme:"B2C");
```
Comment (jmprieur) [inline tests/MultipleAuthSchemes/Controllers/HomeB2CController.cs:27]: the string.Empty is probably not even necessary? is it?

File: tests/Microsoft.Identity.Web.Test.Integration/AcquireTokenForUserIntegrationTests.cs [hunk 4]
```diff
@@ -81,8 +94,32 @@ public AcquireTokenForUserIntegrationTests(WebApplicationFactory<Startup> factor
...
+                    else
+                    {
+                        services.AddDistributedMemoryCache();
+                        services.AddDistributedTokenCaches();
+                    }
+
+                    services.BuildServiceProvider();
+                });
+            })
+                        .CreateClient(new WebApplicationFactoryClientOptions
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test.Integration/AcquireTokenForUserIntegrationTests.cs:119]: There seems to be something funny with spaces here.

**Review by jmprieur — APPROVED** [review]
LGTM.
I've left a few suggestions

File: tests/MultipleAuthSchemes/Controllers/HomeB2CController.cs (hunk 3 above)
Comment (jennyf19) [reply tests/MultipleAuthSchemes/Controllers/HomeB2CController.cs:27]: there is no default for the relative path.

**Pull request merged**